test/starknet/vanilla_space.ts

```diff
@@ -0,0 +1,134 @@
+const PROPOSAL_METHOD = 'propose';
+const VOTE_METHOD = 'vote';
+const GET_PROPOSAL_INFO = 'get_proposal_info';
+const GET_VOTE_INFO = 'get_vote_info';
+const VOTING_DELAY = BigInt(0);
+const VOTING_PERIOD = BigInt(20);
+const PROPOSAL_THRESHOLD = SplitUint256.fromUint(BigInt(1));
+const VITALIK_ADDRESS = BigInt(0xd8da6bf26964af9d7eed9e03e53415d37aa96045);
+
+async function setup() {
+  const vanillaSpaceFactory = await starknet.getContractFactory(
+    './contracts/starknet/space/space.cairo'
+  );
+  const vanillaVotingStategyFactory = await starknet.getContractFactory(
+    './contracts/starknet/strategies/vanilla_voting_strategy.cairo'
+  );
+  const vanillaAuthenticatorFactory = await starknet.getContractFactory(
+    './contracts/starknet/authenticator/authenticator.cairo'
+  );
+  const vanillaAuthenticator = (await vanillaAuthenticatorFactory.deploy()) as StarknetContract;
+  const vanillaVotingStrategy = (await vanillaVotingStategyFactory.deploy()) as StarknetContract;
+  const voting_strategy = BigInt(vanillaVotingStrategy.address);
+  const authenticator = BigInt(vanillaAuthenticator.address);
+  const vanillaSpace = (await vanillaSpaceFactory.deploy({
+    _voting_delay: VOTING_DELAY,
+    _voting_period: VOTING_PERIOD,
+    _proposal_threshold: PROPOSAL_THRESHOLD,
+    _voting_strategy: voting_strategy,
+    _authenticator: authenticator,
+  })) as StarknetContract;
+
+  return {
+    vanillaSpace,
+    vanillaAuthenticator,
+    vanillaVotingStrategy,
+  };
+}
+
+describe('Space testing', () => {
+  it('Simple Vote', async () => {
+    const { vanillaSpace, vanillaAuthenticator, vanillaVotingStrategy } = await setup();
+    const space_contract = BigInt(vanillaSpace.address);
+    const execution_hash = BigInt(1);
+    const metadata_uri = BigInt(2);
+    const proposer_address = VITALIK_ADDRESS;
+    const proposal_id = 1;
+
+    // -- Creates the proposal --
+    {
+      await vanillaAuthenticator.invoke(EXECUTE_METHOD, {
+        to: space_contract,
+        function_selector: BigInt(getSelectorFromName(PROPOSAL_METHOD)),
+        calldata: [proposer_address, execution_hash, metadata_uri],
+      });
+
+      const { proposal_info: proposal_info } = await vanillaSpace.call('get_proposal_info', {
+        proposal_id: proposal_id,
+      });
+
+      // We can't directly compare the `info` object because we don't know for sure the value of `start_block` (and hence `end_block`),
+      // so we compare it element by element (except start_block and end_block for which we simply compare their difference to `VOTING_PERIOD`).
+      expect(proposal_info.proposal.execution_hash).to.deep.equal(execution_hash);
+      expect(proposal_info.proposal.metadata_uri).to.deep.equal(metadata_uri);
+      expect(
+        proposal_info.proposal.end_timestamp - proposal_info.proposal.start_timestamp
+      ).to.deep.equal(VOTING_PERIOD);
+
+      const _for = SplitUint256.fromObj(proposal_info.power_for).toUint();
```

Comment:
I introduced `fromObj` because `proposal_info.power_for` is of type `{low: bigint; high: bigint}` when I would like a `SplitUin256{low: bigint, high: bigint}`. I tried casting by doing `const _for = (proposal_info.power_for as SplitUint256).toUint();` but didn't manage to make it work. Maybe the `.ts` god @bonustrack could help on that :D